src/trace/trace.cpp

```diff
@@ -332,42 +347,53 @@ void Trace::add_lo2s_property(const std::string& name, const std::string& value)
 otf2::writer::local& Trace::sample_writer(pid_t pid, pid_t tid)
 {
     auto name = (boost::format("thread %d") % tid).str();
-    auto location =
-        locations_.emplace(location_ref(), intern(name), location_groups_process_.at(pid),
-                           otf2::definition::location::location_type::cpu_thread);
-    process_comm_groups_.at(pid).add_member(location);
-    return archive()(location);
-}
 
+    // As the tid is unique in this context, create only one writer/location per tid
+    auto location = thread_locations_.emplace(
+        std::piecewise_construct, std::forward_as_tuple(tid),
+        std::forward_as_tuple(location_ref(), intern(name), location_groups_process_.at(pid),
+                              otf2::definition::location::location_type::cpu_thread));
+    process_comm_groups_.at(pid).add_member(location.first->second);
+
+    return archive()(location.first->second);
+}
```

Comment:
Please give the code some space to breath :) (aka insert a newline between the two functions)